addons/project/models/project_milestone.py

```diff
@@ -45,10 +46,15 @@ def _compute_is_deadline_future(self):
 
     @api.depends('task_ids.milestone_id')
     def _compute_task_count(self):
-        task_read_group = self.env['project.task']._read_group([('milestone_id', 'in', self.ids), ('allow_milestones', '=', True)], ['milestone_id'], ['milestone_id'])
-        task_count_per_milestone = {res['milestone_id'][0]: res['milestone_id_count'] for res in task_read_group}
+        task_read_group = self.env['project.task']._read_group([('milestone_id', 'in', self.ids), ('allow_milestones', '=', True)], ['milestone_id', 'is_closed:array_agg(is_closed)'], ['milestone_id'])
+        all_task_count_per_milestone = {}
+        done_task_count_per_milestone = {}
+        for res in task_read_group:
+            all_task_count_per_milestone = {res['milestone_id'][0]: res['milestone_id_count']}
+            done_task_count_per_milestone = {res['milestone_id'][0]: res['is_closed'].count(True)}
         for milestone in self:
-            milestone.task_count = task_count_per_milestone.get(milestone.id, 0)
+            milestone.task_count = all_task_count_per_milestone.get(milestone.id, 0)
+            milestone.done_task_count = done_task_count_per_milestone.get(milestone.id, 0)
```

Comment:
```suggestion
        all_and_done_task_count_per_milestone = {
            group['milestone_id'][0]: (group['milestone_id_count'], group['is_closed'].count(True))
            for group in self.env['project.task']._read_group(
                [('milestone_id', 'in', self.ids), ('allow_milestones', '=', True)],
                ['milestone_id', 'is_closed:array_agg(is_closed)'], ['milestone_id']
            )
        }
        for milestone in self:
            milestone.task_count, milestone.done_task_count = all_and_done_task_count_per_milestone.get(milestone.id, (0, 0))
```
Is a little more straight forward imo :+1: